src/App.tsx

```diff
@@ -20,17 +25,133 @@ class App extends React.Component<{}, State> {
       './img/9.png',
       './img/10.png',
     ],
+    step: 3,
+    frameSize: 3,
+    itemWidth: 130,
+    animationDuration: 1000,
+    infinite: false,
+  };
+
+  handleInputChange = (
+    e: React.ChangeEvent<HTMLInputElement>,
+    key: keyof State,
+  ) => {
+    const { value } = e.target;
+
+    this.setState(prevState => ({
+      ...prevState,
+      [key]: +value,
+    }));
+  };
+
+  handleCheckboxChange = (
+    e: React.ChangeEvent<HTMLInputElement>,
+    key: keyof State,
+  ) => {
+    const { checked } = e.target;
+
+    this.setState(prevState => ({
+      ...prevState,
+      [key]: checked,
+    }));
   };
 
   render() {
-    const { images } = this.state;
+    const {
+      images, itemWidth, frameSize, step, animationDuration, infinite,
+    } = this.state;
 
     return (
       <div className="App">
-        {/* eslint-disable-next-line */}
-        <h1>Carousel with {images.length} images</h1>
+        <h1 data-cy="title" className="title">
+          Carousel with
+          {' '}
+          {images.length}
+          {' '}
+          images
+        </h1>
+
+        <div className="settings">
+          <h2 className="setting__title">Settings</h2>
+
+          <div className="settings__wrapper">
+            <label htmlFor="itemWidth" className="settings__label">
+              Item width:
+            </label>
+            <input
+              type="number"
+              className="settings__input"
+              name="itemWidth"
+              step={10}
+              value={itemWidth}
+              onChange={(e) => this.handleInputChange(e, 'itemWidth')}
```

Comment:
```suggestion
              onChange={this.handleInputChange}
```
you can use event.target.name which will be equal value of name attribute for change proper field in state